#!/usr/bin/env python
PACKAGE = "dynamic_calibration"

from dynamic_reconfigure.parameter_generator_catkin import *
 
gen = ParameterGenerator()

gen.add("roll", double_t, 0, "roll angle", 0, -3.1415926535, 3.1415926535 )
gen.add("pitch", double_t, 0, "pitch angle",0, -3.1415926535, 3.1415926535)
gen.add("yaw", double_t, 0, "yaw angle", 0, -3.1415926, 3.1415926)
gen.add("x", double_t, 0, "translation t_x", 0, -10.1415926535, 10.1415926535)
gen.add("y", double_t, 0, "translation t_y", 0, -10.1415926535, 10.1415926535)
gen.add("z", double_t, 0, "translation t_z", 0, -10.1415926535, 10.1415926535)
gen.add("dis_filter", bool_t, 0, "filter too far pointcloud", False)
gen.add("next", bool_t, 0, "next data", False)
gen.add("get_point", bool_t, 0, "get points for PnP", False)
gen.add("reject_distance", int_t, 0, "image buff", 8, 0, 60)

exit(gen.generate(PACKAGE, "dynamic_calibration", "calibration"))
